### PR TITLE
handleFileUploadComplete updates wrong draft

### PR DESCRIPTION
**Pull request by LeonardJouve**
#### Summary
handleFileUploadComplete updates wrong draft.

If you upload a file on a channel and switch to another channel while the upload is still processing, once the upload completes it updates the draft in the current channel (not the draft where you started uploading your file).

As a result, a new draft is created in the wrong channel and the other channel where you started uploading the file has 1 upload stuck as processing.

This pr fixes this case and improve the test to catch it.

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/advanced_create_post/advanced_create_post.tsx
Comment (devinbinnie) [inline webapp/channels/src/components/advanced_create_post/advanced_create_post.tsx:1000]: I think we might want to rework how `handleDraftChange` here works. What we could do is supply the channelId to it so that we still clear the `saveDraftFrame`, and we don't have to repeat the code that's in `handleDraftChange` here.

**Review by devinbinnie — CHANGES_REQUESTED** [review]
Thanks for the fix @LeonardJouve, seems to be working well :)
I have a comment about how it's designed, but otherwise LGTM.

File: webapp/channels/src/components/advanced_create_post/advanced_create_post.tsx
Comment (LeonardJouve) [reply webapp/channels/src/components/advanced_create_post/advanced_create_post.tsx]: I did the changes. Let me know if this looks good for you : )

**Review by devinbinnie — APPROVED** [review]
Thanks @LeonardJouve, this is looking good now :)

**Comment by LeonardJouve** [comment]
ping

**Comment by LeonardJouve** [comment]
Hi, can I do anything else to help you merge this small fix ? or is it just pending ? :)

**Review by calebroseland — APPROVED** [review]

**Comment by calebroseland** [comment]
Hi @LeonardJouve, apologies—we'll get this moving here, again. Thanks for your patience 🙏

**Comment by yasserfaraazkhan** [comment]
e2e tests are blocked by CI failure

**Comment by yasserfaraazkhan** [comment]
will test once ci is green

**Comment by LeonardJouve** [comment]
Sorry for the wait, I will try my best to do it this week

**Comment by hanzei** [comment]
@LeonardJouve Please let us know if you need any help revolving the test issues.

**Comment by LeonardJouve** [comment]
@hanzei It should be good now ! I updated my fork and fixed the failing test. Let me know if something is missing. :)

**Comment by hanzei** [comment]
@yasserfaraazkhan Would you please give the PR a review?

**Review by yasserfaraazkhan — APPROVED** [review]
changing channel while file upload is in-progess updates the right draft.

**Comment by M-ZubairAhmed** [comment]
Fixes https://github.com/mattermost/mattermost/pull/23983

**Pull request merged**
